qcportal/qcportal/external/scaffold.py

```diff
@@ -81,17 +84,30 @@ def from_json(filename, client):
     else:
         raise ValueError(f"File extension must be json or json.bz2, not {extension[-2]}.{extension[-1]}")
 
-    ds = client.add_dataset(**ds_dict["metadata"])
+    if append:
+        ds = client.add_dataset(**ds_dict["metadata"], existing_ok=True)
+        print("Appending dataset.")
+    else:
+        ds = client.add_dataset(**ds_dict["metadata"])
+        print("Creating new dataset.")
 
-    for _, spec in ds_dict["specifications"].items():
-        ds.add_specification(**spec)
+    for spec_name, spec in ds_dict["specifications"].items():
+        if spec_name not in ds.specifications:
+            ds.add_specification(**spec)
+        else:
+            warnings.warn(f'Specification, {spec_name}, is already in the dataset: {ds_dict["metadata"]["name"]}')
 
     entries = []
     entry_type = ds._entry_type
-    for _, entry in ds_dict["entries"].items():
+    for entry_name, entry in ds_dict["entries"].items():
         if "local_results" in entry:
             del entry["local_results"]
-        entries.append(entry_type(**entry))
+            
+        if entry_name in ds.entry_names:
```

Comment:
I think this should be `if entry_name not in ds.entry_names` ? I was just hunting down some pytest warnings :)